sample-compose/app/src/androidTest/java/co/nimblehq/sample/compose/test/CoroutineTestRule.kt

```diff
@@ -0,0 +1,32 @@
+package co.nimblehq.sample.compose.test
+
+import co.nimblehq.sample.compose.util.DispatchersProvider
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.*
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class CoroutineTestRule(
+    private val testDispatcher: TestDispatcher = UnconfinedTestDispatcher()
+) : TestWatcher() {
+
+    override fun starting(description: Description?) {
+        super.starting(description)
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    override fun finished(description: Description?) {
+        super.finished(description)
+        Dispatchers.resetMain()
+    }
+
+    val testDispatcherProvider = object : DispatchersProvider {
+        override val io: CoroutineDispatcher
+            get() = testDispatcher
+        override val main: CoroutineDispatcher
+            get() = testDispatcher
+        override val default: CoroutineDispatcher
+            get() = testDispatcher
+    }
```

Comment:
Should this be placed before `fun starting(...)`?

Reply:
👍

Reply:
Fixed in https://github.com/nimblehq/android-templates/pull/384/commits/d0a4ce710d805742b5160d44b28f657ed6b8f854